src/components/checkbox-group/checkbox/bl-checkbox.ts

```diff
@@ -0,0 +1,96 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import style from './bl-checkbox.css';
+import type BlCheckboxGroup from '../bl-checkbox-group';
+import { event, EventDispatcher } from '../../../utilities/event';
+
+/**
+ * @tag bl-checkbox
+ * @summary Baklava Checkbox component
+ */
+
+@customElement('bl-checkbox')
+export default class BlCheckbox extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+  private checkboxGroup: BlCheckboxGroup | null;
+
+  @query('#checkbox')
+  checkbox: any;
+
+  @query('#label')
+  labelEl: any;
+
+  @property({ type: String, reflect: true })
+  label = '';
+
+  @property({ type: Boolean, reflect: true })
+  checked = false;
+
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  @property({ type: Boolean, reflect: true })
+  indeterminate = false;
```

Comment:
The `indeterminate` feature should be thought about again. Because when I discuss with @buseselvi she said only the first checkbox in the group checkbox can be `indeterminate`. But when I try I can set any checkbox as `indeterminate`. We shouldn't set as `indeterminate` any checkbox.

Reply:
I changed the logic it as you said. now only the first checkbox in the group can have indeterminate state. but to do this, i add a new property called `allowIndeterminate` to `bl-checkbox`. unfortunately it shows up in docs. I couldn't do it any other way

Reply:
Why we need a logic like "only first checkbox can be `indeterminate`"? I never saw a restriction like this before. Generally indeterminate checkboxes are used in tree-like selections. Parent of a checkbox group can be indeterminate state regarding to selection of sub checboxes. But it doesn't need to be first checbox in the parent group.

Reply:
Actually, I wanted to say only the parent checkbox of the group can be `indeterminate`.